snapcraft/file_utils.py

```diff
@@ -110,6 +110,11 @@ def link_or_copy(source, destination, follow_symlinks=False):
         # symlinks.
         os.link(source_path, destination, follow_symlinks=False)
     except OSError:
+        # If os.link raised an I/O error, it may have left a file behind.
+        # Skip on OSError in case it doesn't exist or is a directory.
+        with suppress(OSError):
```

Comment:
You only get an OSError if it's a directory. You still get FileNotFoundError if it  doesn't exist. You likely want `suppress(FileNotFoundError, OSError)` here.